dev-docs/component-dev-notes.md

```diff
@@ -0,0 +1,30 @@
+### Errors
+
+Components are intended to interface with the registry system - they are registratable objects.
+The intent of this is that the system knows what objects have been registered so it can present to the user on an API level all the components which can be included in a bot configuration.
+However, setting this up does mean adding some definitions in a couple of different places.
+
+### If you are changing a declared API
```

Comment:
Do we want to say "<these> changes do not require an increment to the api". Probably stuff along the lines of "you are purely adding new features not modifying old ones"

Reply:
This seems clearer - updated